tests/class-test-post-bulk-task.php

```diff
@@ -96,4 +96,20 @@ function ( WP_Post $post ): void {
 		$this->assertEquals( 'apple', get_post( $this->post_ids[0] )->post_content );
 		$this->assertEquals( 'banana', get_post( $this->post_ids[1] )->post_content );
 	}
+
+	/**
+	 * Tests getting the current query object.
+	 */
+	public function test_get_query(): void {
+		$query = null;
+
+		( new Bulk_Task( 'test_query_run' ) )->run(
+			[],
+			function ( $_, $__, $post_query ) use ( &$query ): void {
```

Comment:
The array `key` is always the second argument with `array_walk`.

https://www.php.net/manual/en/function.array-walk.php

Reply:
Interesting. Do you think that the array key is a part of the library's "public API"? I can't say it ever occurred to me that the key was available, but that's just me.

If the array key is _not_ a part of the public API, then perhaps we could make it easier on implementers by wrapping `array_walk` with our own function so that the query object is the second parameter passed to the callable, like

```
array_walk( 
	$this->query->posts,
	function ( $post ) use ( $callable ) {
		$callable( $post, $this->query );
	}
);
```

Reply:
It didn't occurred to me we had access to key either, so not officially part of the "public API".

However, I'd like to make it official. That would:

- make the key available if necessary (for any reason);
- would not create a misalignment with the CSV callback, that expects the key as the second parameter:
https://github.com/alleyinteractive/wp-bulk-task/blob/6fc381cab4c0494b7695e97725a465d9007b7a13/src/class-bulk-task.php#L359

Reply:
Makes sense 👍